src/main/java/core/basesyntax/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler operationHandler = operationStrategy
+                    .getHandler(transaction.getOperation());
+            operationHandler.handle(transaction, storage);
+        }
+    }
+
+    public Map<String, Integer> getStorage() {
```

Comment:
The `getStorage()` method exposes the internal storage details, which breaks encapsulation. Consider removing this method or returning a copy of the storage map to prevent direct modifications.